install/kubernetes/cilium/templates/_helpers.tpl

```diff
@@ -165,3 +165,23 @@ Convert a map to a comma-separated string: key1=value1,key2=value2
 {{- end -}}
 {{ join "," $list }}
 {{- end -}}
+
+{{/*
+Enable automatic lookup of k8sServiceHost and k8sServicePort values from the cluster-info ConfigMap (kubeadm-based clusters only)
+*/}}
+{{- define "k8sService" }}
+  {{- if eq .host "auto" }}
+    {{- $configmap := (lookup "v1" "ConfigMap" "kube-public" "cluster-info") }}
+    {{- $kubeconfig := get $configmap.data "kubeconfig" }}
+    {{- range ( split "\n" $kubeconfig) }}
+      {{- if ( contains "server: https://" .) }}
+        {{- $uri := (split "https://" .)._1 | trim }}
+        {{- if eq $.lookup "host" }}{{ (split ":" $uri)._0 | quote }}{{ end }}
+        {{- if eq $.lookup "port" }}{{ (split ":" $uri)._1 | quote }}{{ end }}
+      {{- end }}
+    {{- end }}
+  {{- else }}
+    {{- if eq .lookup "host" }}{{ .host | quote }}{{ end }}
```

Comment:
I think it would be cleaner if we split this out into two separate helpers: `k8sServerHost` and `k8sServerPort`. 

We can then also pass in the full `.` at call sites like this, instead of building and then parsing a custom dict argument:

```
{{ include "k8sServiceHost" . }}
```
or
```
{{ include "k8sServicePort" . }}
```